examples/all-clusters-app/linux/fuzzing-main.cpp

```diff
@@ -84,8 +88,17 @@ extern "C" int LLVMFuzzerTestOneInput(const uint8_t * aData, size_t aSize)
 
     // Ignoring the return value from OnMessageReceived, because we might be
     // passing it all sorts of garbage that will cause it to fail.
-    Server::GetInstance().GetSecureSessionManager().OnMessageReceived(peerAddr, std::move(buf));
 
+    // for TCP we need to have MessageTransportContext
+    if (fuzzedTransportType == Transport::Type::kTcp)
+    {
+        Transport::MessageTransportContext msgContext;
```

Comment:
This isn't a valid MessageTransportContext (has no actual connection pointer) for TCP, so this can cause crashes that would never arise normally.

Reply:
- actually I added `msgContext` and this whole `if condition` because without it `ctxt` below was null and an ASAN crash related to dereferencing the null pointer in `ctxt->conn == nullptr` showed up...I wonder if we should do something about it 
- in the code we seem to protect well against not have an actual connection pointer with `ctxt->conn == nullptr`

- what are your suggestions?

https://github.com/project-chip/connectedhomeip/blob/1d6b0a5815dbaa2a8c46bd5e27648b661b746e8c/src/transport/SessionManager.cpp#L903

Reply:
If this causes crashes that would not normally happen (or code paths that are not valid normally) we could skip TCP testing.

I guess the question is if context.conn is allowed to be null in every situation and is handled or if we rely on it being non-null in some code and if that is fine. 

@pidarped - is TCP transport fuzzable?

Reply:
The context object pointer should be *not null* for the TCP message processing path. 
The context.conn pointer is expected to be a valid pointer normally, but for the purposes of this fuzz testing can be null/garbage I believe. It is only compared with the sessionConnection pointer already stored in the secure session.

Reply:
Sounds good.